app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,80 @@
+class CommentsController < ApplicationController
+  include ApplicationHelper
+  def new
+    @comment = Comment.new
+    @type = case params[:type]
+              when "Answer"
+                Answer.find_by(id: params[:id])
+              when "Question"
+                Question.find_by(id: params[:id])
+              when "Comment"
+                Comment.find_by(id: params[:id])
+              end
+  end
+
+  def create
+    unless logged_in?
+      flash[:error] = "You must be logged in to do that!"
+      redirect_to users_path
+      return
+    end
+    commentable  = case params[:type]
+                   when "Answer"
+                    Answer.find_by(id: params[:id])
+                   when "Question"
+                    Question.find_by(id: params[:id])
+                   when "Comment"
+                    Comment.find_by(id: params[:id])
+                   end
+    attributes = comment_params.merge(author_id: session[:user_id])
+    @comment = commentable.comments.build(attributes)
+    @question = @comment.associated_question
+    flash[:error] = "incorrect comment format" unless @comment.save
```

Comment:
Be careful about assuming specific errors - let the framework tell the user what was wrong